processing/src/main/java/org/apache/carbondata/processing/merger/RowResultMergerProcessor.java

```diff
@@ -126,7 +125,7 @@ public boolean execute(List<RawResultIterator> unsortedResultIteratorList,
       RawResultIterator iterator = null;
       while (index > 1) {
         // iterator the top record
-        iterator = this.recordHolderHeap.poll();
+        iterator = this.recordHolderHeap.peek();
         Object[] convertedRow = iterator.next();
         if (null == convertedRow) {
           index--;
```

Comment:
insert code: this.recordHolderHeap.poll();

Reply:
done